charts/newrelic-infrastructure/templates/daemonset-kubelet.yaml

```diff
@@ -0,0 +1,294 @@
+              value: "false"
+            - name: "NRI_KUBERNETES_KSM_ENABLED"
+              value: "false"
+            - name: "NRI_KUBERNETES_KUBELET_ENABLED"
+              value: "true"
+            - name: "NRI_KUBERNETES_CLUSTERNAME"
+              value: {{ include "newrelic.cluster" . }}
+            - name: "NRI_KUBERNETES_NODENAME"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+            - name: "NRI_KUBERNETES_NODEIP"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "status.hostIP"
+            {{- with .Values.kubelet.extraEnv }}
+            {{- tpl . $ | nindent 12 }}
+            {{- end }}
+          {{- with .Values.kubelet.extraEnvFrom }}
+          envFrom:
+            {{- tpl . $ | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: nri-kubernetes-config
+              mountPath: /etc/newrelic-infra/integrations.d/nri-kubernetes-config.yml
+              subPath: nri-kubernetes-config.yml
+            - mountPath: /var/cache/nr-kubernetes
+              name: tmpfs-cache
+            - mountPath: /tmp
+              name: tmpfs-tmp
+            {{- with .Values.kubelet.extraVolumeMounts }}
+            {{- tpl . $ | nindent 12 }}
+            {{- end }}
+          {{- with .Values.resources }}
+          resources: {{ toYaml . | nindent 12 }}
+          {{- end }}
+        - name: agent
+          image: {{ .Values.images.agent.repository }}:{{ .Values.images.agent.tag }}
+          imagePullPolicy: {{ .Values.images.agent.pullPolicy }}
+          securityContext:
+            {{- if .Values.privileged }}
+            privileged: true
+            {{- else }}
+            runAsUser: {{ .Values.runAsUser }}
+            runAsGroup: {{ .Values.runAsGroup }}
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: true
+            {{- end }}
+          ports:
+            - containerPort: 8003
+          env:
+            - name: "NRIA_OVERRIDE_HOSTNAME_SHORT"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+            - name: "NRIA_HTTP_SERVER_PORT"
+              value: "8003"
+            - name: NRIA_LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  {{- if (include "newrelic.licenseKey" .) }}
+                  name: {{ template "newrelic.fullname" . }}-config
+                  key: license
+                  {{- else }}
+                  name: {{ include "newrelic.customSecretName" . }}
+                  key: {{ include "newrelic.customSecretLicenseKey" . }}
+                  {{- end }}
+            {{- with .Values.eventQueueDepth }}
+            - name: "NRIA_EVENT_QUEUE_DEPTH"
+              value: {{ .| quote }}
+            {{- end }}
```

Comment:
This affects how the agent sends data so I believe it should be in all sidecars. Thankfully I don't really see a reason to have it per-sidecar, we can probably share the value for all.